packages/hash/README.md

```diff
@@ -16,7 +16,8 @@ HASH is an open-source, data-centric, all-in-one workspace. HASH combines a rich
 
 To run HASH locally, please follow these steps:
 
-1.  Make sure you have, [Git](https://git-scm.com), [Node LTS](https://nodejs.org), [Yarn Classic](https://classic.yarnpkg.com) and [Docker](https://docs.docker.com/get-docker/):
+1.  Make sure you have, [Git](https://git-scm.com), [Node LTS](https://nodejs.org), [Yarn Classic](https://classic.yarnpkg.com) and [Docker](https://docs.docker.com/get-docker/).
+    Run each of these version commands and make sure the output is expected:
 
     ```sh
     git --version
```

Comment:
I'd rather we didn't mention version requirements which are not in fact requirements, but just ways of people checking they have git and Docker (Node does require 16+ from memory) - this makes it a more arduous path to actually run the thing since we're specifying the latest versions.

Can we take this opportunity to remove git at least? Maybe also yarn and Docker, since presumably we don't need any particular version of those.

Reply:
If we say ‘check git, Docker and Yarn’, the immediate question someone might have is ‘how’? Running `xyz --version` just makes it trivial to confirm that everything is ready. If we a worried that folks we older but acceptable versions get confused, we can lower minimal versions.

If we can give users confidence in their environment without asking them to run `xyz --version`, that can work too. I’m just not sure how else to do this.

Reply:
In theory, we can create some starter package for this:

```sh
npx some-hash-cli check-environment

## All good!

## You need to download these tools to get started:
##
## - Docker: https://docs.docker.com/get-docker/
## - Yarn: https://classic.yarnpkg.com/en/docs/install#mac-stable
##
## Please run this command again when you are done
```

That’s not a quick fix though 🙂

Reply:
> Running xyz --version just makes it trivial to confirm that everything is ready

It doesn't when we have specified very recent versions that means that most people are going to have to pointlessly upgrade something to run this, potentially being put off using it.

How about `--help` or equivalent for each?

Reply:
We can’t use `--help` in every case because the versions of Node and Docker do matter (older Docker versions do not support BUILDKIT and may miss some docker-compose features).

The output of `--version` is always short, while `--help` floods the terminal. It won’t be practical to paste all commands in bulk and quickly check the results.

If we don’t want people to make unnecessary upgrades, we can pick the oldest supported version of tools for which we don’t need the latest features. Here is the breakdown:

`git --version` – We mention 2.34 (Nov 21), but can probably use 2.17 (April 2018) - [Wikipedia](https://en.wikipedia.org/wiki/Git)

`node --version` – I guess we need to keep 16.13 because it’s the smallest Node 16 LTS. It was released in Oct 2021. We could add Node 14 LTS (Oct 2020), but do we want to test everything for multiple Node versions at this early stage? I doubt that it’s worth it.

`yarn --version` – We use 1.22, which was released in Feb 2020 – [npm](https://www.npmjs.com/package/yarn?activeTab=versions). Latest 1.21 was released in Nov 2019 and I’m pretty sure it’s unsupported.

`docker --version` – We refer to 20.10. 20.10.0 was released in December 2020 - [GitHub](https://github.com/moby/moby/releases/tag/v20.10.0). Docker Desktop versions are pretty much evergreen.

`docker-compose --version` – We refer to 2.2 (2.2.0 is from Nov 2022 – [GitHub](https://github.com/docker/compose/releases/tag/v2.2.0)) I’m not sure we can use anything older than 2.0.0, which is from Sep 2021 – [GitHub](https://github.com/docker/compose/releases/tag/v2.0.0).

The only opportunity I see is for `git` (2.34 → 2.17). WDYT?

Reply:
It’d be cool to be able to use something like `npx` [`envinfo`](https://www.npmjs.com/package/envinfo), but with specific requirements:

```sh
npx some-magic-command git@>=1.17 node@>16.13 ...
```

If some requirement is not met, the command outputs install instructions for your OS. I haven’t found anything on npm, but I won’t be be surprised it is there!

Reply:
Here are my versions which I've had no trouble with (probably overdue an upgrade, but they work!):
- yarn: `1.16.0`
- docker-compose: `1.29.2`

Node and Docker are up to date.

Reply:
So I suggest we could go for those, along with the git suggestion, as they're known good and will interrupt less people with upgrading.

Reply:
I share Ciaran's aims and sense of priority re: ensuring as low barrier/minimally frictional onboarding as possible (as much as I also want the world to run on the latest stable version of every package this is out of scope and competes with our main aim here).

Reply:
I lowered min versions as suggested: https://github.com/hashintel/hash/pull/190/commits/ddd28efedf37ed22d7520f0159013f445c9aa2c1

Still not sure what’s the right trade-off between initial ‘accessibility’ and possible issues down the line. For example, [Yarn has known vulnerabilities in versions <1.22](https://www.cvedetails.com/vulnerability-list/vendor_id-19880/product_id-57432/Yarnpkg-Yarn.html) and we also don’t know to what extent it supports Yarn Workspaces. Happy to follow your advice for now, but I still wish we had something like this:

```sh
npx some-magic-command x@>=1.1 y@>=42.0 z@*

## - `x` is missing, please install it via http://example.com/x 
## - `y` v20 is installed, but >=42.0 is expected. You can upgrade via http://y.example.org
```